vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/test/grid-connector-tree.test.ts

```diff
@@ -40,28 +41,60 @@ describe('grid connector - tree', () => {
     expect(getBodyCellText(grid, 2, 0)).to.equal('foo bar baz');
   });
 
-
   it('should not update rows each time child items are set', async () => {
     // Add an expanded root item
-    const rootItems = [{ key: '0', name: 'foo', children: true }];
-    setRootItems(grid.$connector, rootItems);
-    expandItems(grid.$connector, rootItems);
-    
+    const rootItem = { key: '0', name: 'foo', children: true };
+    setRootItems(grid.$connector, [rootItem]);
+    expandItems(grid.$connector, [rootItem]);
+
     const spy = sinon.spy(grid, '__updateVisibleRows');
 
     // Add an expanded child item
-    const childItems = [{ key: '1', name: 'foo bar', children: true }];
-    expandItems(grid.$connector, childItems);
-    setChildItems(grid.$connector, '0', childItems);
+    const childItem = { key: '1', name: 'foo bar', children: true };
+    expandItems(grid.$connector, [childItem]);
+    setChildItems(grid.$connector, rootItem, [childItem]);
 
     // Add a grandchild item
     const grandchildItems = [{ key: '2', name: 'foo bar baz' }];
-    setChildItems(grid.$connector, '1', grandchildItems);
+    setChildItems(grid.$connector, childItem, grandchildItems);
 
     await nextFrame();
     await nextFrame();
-    
+
     // This number may come down from further optimization
     expect(spy.callCount).to.equal(3);
   });
+
+  it('should not compute column auto-width prematurely', async () => {
```

Comment:
This verifies that "ensureSubCacheQueue" is checked [here](https://github.com/vaadin/flow-components/blob/5949537412cc0dabe1f3b7a57dee5faa409a1f32/vaadin-grid-flow-parent/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js#L47)